src/common/exec_utils.h

```diff
@@ -594,11 +594,11 @@ inline nnvm::Graph AssignContext(nnvm::Graph g,
   const auto& assigned_device = g.GetAttr<nnvm::DeviceVector>("device");
 
   exec::ContextVector vcontext;
-  for (size_t i = 0; i < assigned_device.size(); ++i) {
-    if (assigned_device[i] == -1) {
+  for (int context : assigned_device) {
```

Comment:
Why you need the type explicitly?

Reply:
I tend to include it in cases where it's not already obvious.